psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1158,6 +1158,56 @@ void ESPPropCalc::compute_field_over_grid(bool print_output) {
     fclose(gridout);
 }
 
+SharedMatrix ESPPropCalc::compute_field_over_grid_in_memory(SharedMatrix input_grid) const {
+    // We only want a plain matrix to work with here:
+    if (input_grid->nirrep() != 1) {
+        throw PSIEXCEPTION("ESPPropCalc only allows \"plain\" input matrices with, i.e. nirrep == 1.");
+    }
+    if (input_grid->coldim() != 3) {
+        throw PSIEXCEPTION("ESPPropCalc only allows \"plain\" input matrices with a dimension of N (rows) x 3 (cols)");
+    }
+
+    int number_of_grid_points = input_grid->rowdim();
+    SharedMatrix output = std::make_shared<Matrix>("Output E field", number_of_grid_points, 3);
+
+    std::shared_ptr<Molecule> mol = basisset_->molecule();
+
+    SharedMatrix Dtot = wfn_->matrix_subset_helper(Da_so_, Ca_so_, "AO", "D");
+    if (same_dens_) {
+        Dtot->scale(2.0);
+    } else {
+        Dtot->add(wfn_->matrix_subset_helper(Db_so_, Cb_so_, "AO", "D beta"));
+    }
+
+    std::shared_ptr<ElectricFieldInt> field_ints(dynamic_cast<ElectricFieldInt*>(wfn_->integral()->electric_field()));
+
+    int const nbf = basisset_->nbf();
+
+    std::vector<SharedMatrix> intmats;
+    intmats.push_back(std::make_shared<Matrix>("Ex integrals", nbf, nbf));
+    intmats.push_back(std::make_shared<Matrix>("Ey integrals", nbf, nbf));
+    intmats.push_back(std::make_shared<Matrix>("Ez integrals", nbf, nbf));
+
+    bool convert = mol->units() == Molecule::Angstrom;
+
+    for (int i = 0; i < number_of_grid_points; ++i) {
```

Comment:
Instead of looping over all grid points, you could adapt [this](https://github.com/psi4/psi4/blob/64915980a2f0e6ea2a65d3e2cf16b3ec4488fe0d/psi4/src/psi4/libmints/mintshelper.cc#L1719) method which computes electric field values at all given `coords` using a functor. Then the for-loop would just be needed to form the nuclear contribution of the electric field.

Reply:
Thanks for the suggestion! This should now be changed in 3850718.